cmake/options.cmake

```diff
@@ -114,4 +114,6 @@ if (DEFINED RPMSG_BUFFER_SIZE)
   add_definitions( -DRPMSG_BUFFER_SIZE=${RPMSG_BUFFER_SIZE} )
 endif (DEFINED RPMSG_BUFFER_SIZE)
 
+option (WITH_DOC "Build with documentation" ON)
```

Comment:
There are a lot of warnings during doc generation. the function and structure headers do not respect the doxygen syntax.
Please deactive the doc generation by default, until it is fixed

Reply:
@arnopo I will see if I can find a way to disable these warnings.

Reply:
@arnopo I disabled warnings - please let me know if this is suitable.

Reply:
I still have warnings. Here is the script I'm using
```
#!/bin/bash
 
dirs_to_check=$PWD
echo "################### build libmetal #####################"
cd libmetal
rm -r build
cmake . -Bbuild -DCMAKE_C_FLAGS="-Wall -Wextra -Wshadow -Wunused-but-set-variable" || exit 1
cd build
make || exit 1

cd $dirs_to_check

echo "################### build openamp #####################"
cd open-amp
rm -r build*
cmake . -DCMAKE_C_FLAGS="-Wall -Wextra -Wshadow -Wunused-but-set-variable" -DWITH_APPS=on -DWITH_DCACHE_BUFFERS=on -DWITH_PROXY=on -Bbuild -DCMAKE_INCLUDE_PATH="../libmetal/build/lib/include" -DCMAKE_LIBRARY_PATH="../libmetal/build/lib" || exit 1
cd build
make install  DESTDIR=./bin || exit 1
```
with following project hierarchy:
```
openamp_project 
|_ open-amp
|_ libmetal
```
More than that  I would prefer to not  mask warnings, we could forget to re-enable the option.
By setting` WITH_DOC` to OFF by default this just keep actual state of the doc generation. 
then to force the doc generation we have just to add `-DWITH_DOC=on` option in the cmake command
`cmake . -DCMAKE_C_FLAGS="-Wall -Wextra -Wshadow -Wunused-but-set-variable" -DWITH_APPS=on -DWITH_DCACHE_BUFFERS=on -DWITH_PROXY=on -Bbuild -DWITH_DOC=on -DCMAKE_INCLUDE_PATH="../libmetal/build/lib/include" -DCMAKE_LIBRARY_PATH="../libmetal/build/lib" `

Reply:
@arnopo I disabled docs generation.  I don't know where all these "file renamed without changes" commits are coming from.  I didn't rename any files (and they aren't renamed) and I only staged the one cmake file for the latest commit.